dbt/include/databricks/macros/python.sql

```diff
@@ -0,0 +1,77 @@
+{% macro databricks__py_write_table(compiled_code, target_relation) %}
+{{ compiled_code }}
+# --- Autogenerated dbt materialization code. --- #
+dbt = dbtObj(spark.table)
+df = model(dbt, spark)
+
+import pyspark
+
+{{ py_try_import('pandas', 'pandas_available') }}
+{{ py_try_import('pyspark.pandas', 'pyspark_pandas_api_available') }}
+{{ py_try_import('databricks.koalas', 'koalas_available') }}
+
+# preferentially convert pandas DataFrames to pandas-on-Spark or Koalas DataFrames first
+# since they know how to convert pandas DataFrames better than `spark.createDataFrame(df)`
+# and converting from pandas-on-Spark to Spark DataFrame has no overhead
+  
+if pandas_available and isinstance(df, pandas.core.frame.DataFrame):
+    if pyspark_pandas_api_available:
+        df = pyspark.pandas.frame.DataFrame(df)
+    elif koalas_available:
+        df = databricks.koalas.frame.DataFrame(df)
+
+# convert to pyspark.sql.dataframe.DataFrame
+if isinstance(df, pyspark.sql.dataframe.DataFrame):
+    pass  # since it is already a Spark DataFrame
+elif pyspark_pandas_api_available and isinstance(df, pyspark.pandas.frame.DataFrame):
+    df = df.to_spark()
+elif koalas_available and isinstance(df, databricks.koalas.frame.DataFrame):
+    df = df.to_spark()
+elif pandas_available and isinstance(df, pandas.core.frame.DataFrame):
+    df = spark.createDataFrame(df)
+else:
+    msg = f"{type(df)} is not a supported type for dbt Python materialization"
+    raise Exception(msg)
+
+writer = (
+    df.write
+        .mode("overwrite")
+        .option("overwriteSchema", "true")
+{{ py_get_writer_options()|indent(8, True) }}
+)
+
+writer.saveAsTable("{{ target_relation }}")
+{% endmacro %}
+
+{%- macro py_get_writer_options() -%}
+{%- set location_root = config.get('location_root', validator=validation.any[basestring]) -%}
```

Comment:
Can we set a default of `None` or `""` here? Or are we now requiring users to always include a location root in their Python models?

This is important, I think, because doing so will break backwards compatibility and we'll need to make a major version change.

Reply:
It already defaults to Jinja none.  No changes required of users.  That's part of what is ensured by running test_python.py, that things continue to build even if none of the new variables are provided.